bump_version

```diff
@@ -0,0 +1,189 @@
+#!/usr/bin/env bash
+
+# bump_version - increase the shared version and generate changelogs
+
+set -o errexit
+set -o pipefail
+
+usage() {
+    echo -e "bump_version - increase the shared version and generate changelogs"
+    echo -e ""
+    echo -e "Usage:"
+    echo -e " $ bump_version [-b/--web-branch <web_branch>] <new_version>"
+    echo -e ""
+    echo -e "The web_branch defaults to the same branch name as the current main branch."
+    echo -e "This helps facilitate releases where both branches would be called release-X.Y.Z"
+    echo -e "and would already be created before running this script."
+}
+
+if [[ -z $1 ]]; then
+    usage
+    exit 1
+fi
+
+shared_version_file="./SharedVersion.cs"
+
+# Parse branch option
+if [[ $1 == '-b' || $1 == '--web-branch' ]]; then
+	web_branch="$2"
+	shift 2
+else
+    web_branch="$( git branch 2>/dev/null | sed -e '/^[^*]/d' -e 's/* \(.*\)/\1/' )"
+fi
+
+# Initialize submodules
+git submodule update --init --recursive
+
+# configure branch
+pushd MediaBrowser.WebDashboard/jellyfin-web
+git fetch --all
+git checkout origin/${web_branch} || {
+    echo "jellyfin-web branch ${web_branch} is invalid."
+    exit 1
+}
+popd
+
+new_version="$1"
+
+# Parse the version from the AssemblyVersion
+old_version="$(
+    grep "AssemblyVersion" ${shared_version_file} \
+        | sed -E 's/\[assembly: ?AssemblyVersion\("([0-9\.]+)"\)\]/\1/' \
+        | sed -E 's/.0$//'
+)"
+
+# Set the shared version to the specified new_version
+old_version_sed="$( sed 's/\./\\./g' <<<"${old_version}" )" # Escape the '.' chars
+sed -i "s/${old_version_sed}/${new_version}/g" ${shared_version_file}
+
+declare -a pr_merges_since_last_master
+declare changelog_string_github
+declare changelog_string_deb
+declare changelog_string_yum
+
+# Build up a changelog from merge commits
+for repo in ./ MediaBrowser.WebDashboard/jellyfin-web/; do
+    last_master_merge_commit=""
+    pr_merges_since_last_master=()
+    git_show_details=""
+    pull_request_id=""
+    pull_request_description=""
+    changelog_strings_repo_github=""
+    changelog_strings_repo_deb=""
+    changelog_strings_repo_yum=""
+
+    case $repo in
+        *jellyfin-web*)
+            repo_name="jellyfin-web"
+        ;;
+        *)
+            repo_name="jellyfin"
+        ;;
+    esac
+
+    pushd ${repo}
+
+    # Find the last release commit, so we know what's happened since
+    last_master_branch="master-${old_version}" # TODO: Set to `release-` for next release, so we can properly parse both repos with the new branch standard
+    last_master_merge_commit="$(
+        git log --merges --pretty=oneline \
+            | grep -F "${last_master_branch}" \
+            | awk '{ print $1 }' \
+            || true # Don't die here with errexit
```

Comment:
If I don't, and the "last master merge" isn't found, it errexits. However I do checking on that in the next few lines and exit nicely if that's the case rather than just bailing here.